JPS_VIRTUALSENSOR/AermodAgent/src/main/java/com/cmclinnovations/aermod/Buildings.java

```diff
@@ -1208,7 +1208,7 @@ public void getProperties() throws org.apache.jena.sparql.lang.sparql_11.ParseEx
         // Remove IRIs of pollutant sources
         for (int i = 0; i < BuildingIRIQueryResult.length(); i++){
             String buildingIRI = BuildingIRIQueryResult.getJSONObject(i).getString("cityObject");
-            if (!StackIRIString.contains(buildingIRI.replace("cityobject", "building"))) {
+            if (!StackIRIString.contains(buildingIRI.replace("cityobject", "cityfurniture"))) {
```

Comment:
is this change intentional in this branch?

Reply:
Yes that change is intentional. For JurongIsland, we use the cityfurniture graph to get the height and diameter of pollutant sources. After searching for buildings in a bounding box, we get IRIs from the cityobject graph. To remove the pollutant emitting points from the results list, we need to replace cityobject with cityfurniture when checking if the IRI already exists in the list of pollutant emitting points.

Reply:
If I understand this correctly, this change needs to be committed together with following changes:
1) city gml namespace as input
2) aermod agent querying emission sources directly

Hence I don't think this should be in this branch because it interferes with the functionality of this branch, correct me if I'm wrong

Reply:
I wouldn't say it interferes with the functionality of this branch since all the new functionality is within the JurongIslandInputAgent folder.

Reply:
Are you sure? I mean if you merge this into the 1461-create-an-aermod-agent-that-retrieves-building-related-information-from-the-knowledge-graph branch but still query the emissions using the old way won't the results change?

Reply:
Actually, the query has been updated  in the 1461-create-an-aermod-agent-that-retrieves-building-related-information-from-the-knowledge-graph branch itself. Previously, I queried the OCGML IRIs of buildings from the same namespace that contains the data for emissions and OCGML IRIs of pollutant sources. I removed the call to the method in BuildingsQueryClient that executes that query. Instead, I now use the search within the bounding box to get buildings near pollutant emitting points.

Reply:
if that's the case make sure to increment the version number before merging